src/main/java/gift/service/ProductService.java

```diff
@@ -0,0 +1,51 @@
+package gift.service;
+
+import gift.dto.ProductRequestDto;
+import gift.entity.Product;
+import gift.repository.ProductRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    public List<Product> findAll(){
+        return productRepository.findAll();
+    }
+
+    public Optional<Product> findById(Long id){
+        return productRepository.findById(id);
+    }
+
+    public Product create(ProductRequestDto dto){
+        return productRepository.save(
+                new Product(null, dto.getName(), dto.getPrice(), dto.getImageUrl()));
+    }
+
+    public Optional<Product> update(Long id, ProductRequestDto dto){
+        if(productRepository.findById(id).isEmpty()){
+            return Optional.empty();
+        }
+
+        Product product = new Product(id, dto.getName(), dto.getPrice(), dto.getImageUrl());
+        productRepository.update(id, product);
+        return Optional.of(product);
```

Comment:
업데이트된 Product를 신규로 만들어서 갈아치우기보다는 이미 저장된 product를 가지고와서 객체를 update 해주면 어떨까요? 🤔

```suggestion
        Product product = productRepository.findById(id)
            .orElseThrow(() -> new NoSuchElementException("예외 메시지...."));

        product.update(dto.getName(), dto.getPrice(), dto.getImageUrl());
```